tests_v2/unit/app/routes/datasets/test_query.py

```diff
@@ -823,6 +838,120 @@ async def test_query_batch_uri(
     assert response.json()["status"] == "success"
 
 
+@pytest.mark.asyncio
+async def test_query_dataset_json_with_overrides(
+    generic_raster_version,
+    geostore,
+):
+    dataset_name, version_name, _ = generic_raster_version
+    geostore_common = await get_geostore(geostore, GeostoreOrigin.rw)
+    sql = "SELECT sum(area__ha) FROM data"
+
+    with patch(
+        "app.routes.datasets.queries._query_raster", return_value={"data": []}
+    ) as mock_query_raster:
+        result = await _query_dataset_json(
+            dataset_name, version_name, sql, geostore_common, {dataset_name: "v2"}
+        )
+
+        mock_query_raster.assert_awaited_once()
+        args = mock_query_raster.await_args
+        assert args[1]["version_overrides"] == {dataset_name: "v2"}
+        assert result == []
+
+
+@pytest.mark.asyncio
+async def test_query_raster_with_overrides(
+    generic_raster_version,
+    geostore,
+):
+    dataset_name, version_name, _ = generic_raster_version
+    asset = await get_default_asset(dataset_name, version_name)
+    geostore_common = await get_geostore(geostore, GeostoreOrigin.rw)
+    sql = "SELECT sum(area__ha) FROM data"
+
+    with patch(
+        "app.routes.datasets.queries._query_raster_lambda", return_value={"data": []}
+    ) as mock_raster_query_lambda:
+        result = await _query_raster(
+            dataset_name,
+            asset,
+            sql,
+            geostore_common,
+            version_overrides={dataset_name: "v2"},
+        )
+
+        mock_raster_query_lambda.assert_awaited_once()
+        args = mock_raster_query_lambda.await_args
+        assert args[0][5] == {dataset_name: "v2"}
+        assert result == {"data": []}
+
+
+@pytest.mark.asyncio
+async def test_query_raster_lambda_with_overrides(
+    generic_raster_version,
+    geostore,
+):
+    dataset_name, _ = generic_raster_version
```

Comment:
There are actually three things in here, hence this failure: https://github.com/wri/gfw-data-api/actions/runs/13687801477/job/38275023712#step:3:5005

Reply:
Ah, I had accidentally removed that last minute due to flake8 message, fixed!